### PR TITLE
Use subprocess to kill process and fix the gate

### PR DESCRIPTION
**Pull request by sileht**
This change uses subprocess to kill processes.
This avoid to have zombie processes by closing stdin/stdout, flushing
buffer and retrieving the dead process status.

For mysql, we don't need pid-file since we don't use mysql_safe.

### CODE REVIEW

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 7286a4a502c65b7222dfb8f2a7d8cd6c33b48776

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 9beceab0db91b9ff5c7e74aa3fdcae52b8bb817c

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 4c4b47d1cbc876520b707c178a57ed938a74dfd6

**Comment by sileht** [comment]
Pull-request updated, HEAD is now de1c6f83432745ff579fef988a74854d7b36571e

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 61edf425d0589998d876324d08e0ea8d5f38e13c

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 14395b3ff894c90d0599f72c94091a42ede36fc4

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 61edf425d0589998d876324d08e0ea8d5f38e13c

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 6561d28da21cf13068e36bbe1a7ae11fc5148018

File: pifpaf/drivers/__init__.py [hunk 1]
```diff
@@ -63,33 +63,21 @@ def putenv(self, key, value, raw=False):
...
-    def _kill(self, pid, sig=signal.SIGTERM):
-        os.kill(pid, sig)
-
-        # Wait 10 seconds max
+    def _kill(self, process):
+        pgrp = os.getpgid(process.pid)
+        process.terminate()
         try:
-            self._wait(pid)
+            tenacity.retry(
```
Comment (jd) [inline pifpaf/drivers/__init__.py]: you could define a `Retrying` object once and for all on the class and then call it here.
It'd be cheaper. :) not a big deal though:

```
_WAIT_FOR_POLL_NONE = tenacity.Retrying(wait=tenacity.wait_fixed(1),
                 stop=tenacity.stop_after_attempt(10),
                 retry=tenacity.retry_if_result(lambda ret: ret is None))

[...]

try:
    self._WAIT_FOR_POLL_NONE(process.poll)
except tenacity.RetryError:
    [...]
```

File: pifpaf/drivers/__init__.py [hunk 2]
```diff
@@ -25,6 +25,7 @@
 import fixtures
 import jinja2
 import six
+import xattr
```
Comment (jd) [inline pifpaf/drivers/__init__.py]: xattr is not available on Windows so it's a bit problematic to import it without a condition.

That was the main reason of having the `ceph` flavor.

File: pifpaf/drivers/__init__.py (hunk 1 above)
Comment (sileht) [inline pifpaf/drivers/__init__.py]: I wonder why I changed this...

File: pifpaf/drivers/__init__.py (hunk 2 above)
Comment (sileht) [inline pifpaf/drivers/__init__.py]: fixed

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 7828eb9e8e6e606881a445e9d65e938be5ebe9a3

**Pull request merged**